File: C/2283-check-if-number-has-equal-digit-count-and-digit-value/main.go
package main

func digitCount(num string) bool {
	m := map[int]int{}
	for _, b := range num {
		m[int(b-('0'))]++
	}

	for i := 0; i < len(num); i++ {
		if m[i] != int(num[i]-'0') {
			return false
		}
	}
	return true
}

File: C/2283-check-if-number-has-equal-digit-count-and-digit-value/main.rs
struct Solution {}

impl Solution {
  pub fn digit_count(num: String) -> bool {
    let mut digits: [i32; 10] = [0; 10];
    num
      .as_bytes()
      .iter()
      .for_each(|b| digits[(b - '0' as u8) as usize] += 1);

    num
      .as_bytes()
      .iter()
      .enumerate()
      .map(|(i, d)| digits[i] == (*d - '0' as u8) as i32)
      .all(|x| x)
  }
}
